### ISSUE TITLE
LM failure kills everyone, though dead capsule crew brings the ship home

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#287]
In the same lunar landing mission I mentioned in #286, the LM lifted off but couldn't make it up to the capsule, made an emergency landing and the crew died on the surface.  Yet...the guy up in the capsule died too: 
![Primary crew dead](https://user-images.githubusercontent.com/3118014/67624116-acc7fa80-f7e1-11e9-94ef-2ad8d4b33a35.png)

Yet, he managed to fly the capsule successfully back to Earth, only to be dead again on arrival.
![All dead](https://user-images.githubusercontent.com/3118014/67624131-d08b4080-f7e1-11e9-8652-5ef8ee6a7d64.png)

**Comment by peyre** [comment]
Also, it looks like the yellow bar between Lovell and Aldrin was drawn in the wrong place.

**Comment by rnyoakum** [comment]
I'm also impressed that they made it all the way there in 1-2 days (Duration A).

**Comment by peyre** [comment]
Like Hudson in Aliens: "We're on an express elevator to hell, going down."

**Comment by peyre** [comment]
Incidentally, this isn't exactly an isolated incident.  There seems to be trouble with the handling of failures with the LM, at least on the lunar landing itself.  Killing the LM crew often (always?) kills the capsule crew as well; I suspect the game is used to handling the crew as a unit and just isn't making an exception when some of them are in the LM (though it handles EVA incidents fine).

**Comment by peyre** [comment]
This is interesting.  Seems the logic works correctly if the failure happens on the LM's descent to the surface.  I watched my LM crash on the Moon this morning, and it only killed the two who were aboard.  When I came into my next turn the crew was gone, and the capsule pilot who survived was still alive, though he was planning to retire (and I don't blame him).

![LM Destroyed](https://user-images.githubusercontent.com/3118014/69012355-cc0dff80-0929-11ea-9f61-cd5a53e88e05.png)

**Comment by peyre** [comment]
Ha, just had this one again.

![Add dead due to LM](https://user-images.githubusercontent.com/3118014/79644827-443e5280-8160-11ea-8b17-9715d4475ad4.png)

**Comment by rnyoakum** [comment]
Duplicate of #476.

The main point of distinction is that selecting a person to walk on the moon modifies the `EVA` field, which impacted how the LM failure decided who to kill. I'm closing this issue and leaving the other open, since it contains [more complete commentary](https://github.com/raceintospace/raceintospace/issues/476#issuecomment-700362726) on what is happening.

**Comment by peyre** [comment]
Oh no, I just experienced this again!

![Lunar Liftoff Explosion](https://user-images.githubusercontent.com/3118014/97086291-29e5a580-15d7-11eb-8b1c-8f448e266f9a.png)

**Comment by peyre** [comment]
I'm running the mission over and over to test for #476.  I just experienced this again - same failure, same result.  Maybe this failure code, specifically, is handled wrong?

**Comment by peyre** [comment]
Same one again!  Well, at least it's consistent.

**Comment by rnyoakum** [comment]
I know what's going on here!

Mission errors are classified by FName - this is what determines the general type of error. The FName is what groups Orbital Insertion Burn errors separately from Lunar EVA errors. It also distinguishes errors by hardware, in some cases. Hence, the FName "F009" refers to the set of errors that can occur on a Trans-Lunar Injection, "F019" Lunar Liftoff errors (LM), "F119" Lunar Liftoff (capsule) errors, and so forth.

Anyways, within the past week I noticed what looked like a bunch of off-by-one errors in the `MissionSteps()` function where the FName values are generated for each mission step. As one consequence, for example, **all** Lunar Liftoff steps would be assigned the F119 error set.

Using the F119 error set would generate code 3 errors for a step failure, instead of code 31 errors. A code 3 error kills the entire flight crew, while a code 31 error only kills the LM crew. That appears to be the situation here.

So, thank you. I was concerned that I was misunderstanding the FName generation, and planned to do a lot of testing today. I actually wrote the fix last night, so it should be pushed later today :)
